src/networking.c

```diff
@@ -317,27 +317,15 @@ int prepareClientToWrite(client *c) {
  * Low level functions to add more data to output buffers.
  * -------------------------------------------------------------------------- */
 
-/* Attempts to add the reply to the static buffer in the client struct.
- * Returns the length of data that is added to the reply buffer.
+/* Adds reply to the static buffer in the client struct.
  *
  * Sanitizer suppression: client->buf_usable_size determined by
  * zmalloc_usable_size() call. Writing beyond client->buf boundaries confuses
  * sanitizer and generates a false positive out-of-bounds error */
 REDIS_NO_SANITIZE("bounds")
-size_t _addReplyToBuffer(client *c, const char *s, size_t len) {
-    size_t available = c->buf_usable_size - c->bufpos;
-
-    /* If there already are entries in the reply list, we cannot
-     * add anything more to the static buffer. */
-    if (listLength(c->reply) > 0) return 0;
-
-    size_t reply_len = len > available ? available : len;
+static inline void _addReplyToBuffer(client *c, const char *s, size_t reply_len) {
```

Comment:
after this changes this method becomes dangerous (no longer judging whether there is enough space), and this method is just being used by `_addReplyToBufferOrList()`, maybe we can eliminate this method and move these two lines to there.